emukit/bayesian_optimization/acquisitions/expected_improvement.py

```diff
@@ -73,3 +73,73 @@ def evaluate_with_gradients(self, x: np.ndarray) -> Tuple:
     def has_gradients(self) -> bool:
         """Returns that this acquisition has gradients"""
         return isinstance(self.model, IDifferentiable)
+
+
+
+class IntegratedExpectedImprovement(Acquisition):
+
+    def __init__(self, model: Union[IModel, IDifferentiable, IPriorHyperparameters], jitter: float = float(0),
+                 n_samples = 10) -> None:
+        """
+        This acquisition computes for a given input the improvement over the current best observed value in
+        expectation. This function integrates over hyper-parameters the model by computing the  average of the
+        expected improvements for all samples. For more information see:
+
+        Efficient Global Optimization of Expensive Black-Box Functions
+        Jones, Donald R. and Schonlau, Matthias and Welch, William J.
+        Journal of Global Optimization
+
+        :param model: model that is used to compute the improvement.
+        :param jitter: parameter to encourage extra exploration.
+        """
+
+        self.model = model
+        self.jitter = jitter
+        self.n_samples = n_samples
+        self.samples = self.model.generate_hyperparameters_samples(n_samples)
+
+    def evaluate(self, x: np.ndarray) -> np.ndarray:
+        """
+        Computes the integrated Expected Improvement with respect to the hyper-parameters of the model. Averages the
+        improvement for all the samples.
+
+        :param x: points where the acquisition is evaluated.
+        :return: numpy array with the integrated expected improvement at the points x.
+        """
+
+        if x.ndim == 1: x = x[None, :]
+        improvement = 0
+
+        for sample in self.samples:
+            self.model.fix_model_hyperparameters(sample)
+            acquisition = ExpectedImprovement(self.model, self.jitter)
+            improvement += acquisition.evaluate(x)
+
+        return improvement/self.n_samples
+
+    def evaluate_with_gradients(self, x: np.ndarray) -> Tuple:
+        """
+        Computes the Expected Improvement and its derivative integrating over the hyper-parameters of the model
+
+        :param x: locations where the evaluation with gradients is done.
+        :return: tuple containing numpy arrays with the integrated expected improvement at the points x
+        and its gradient.
+        """
+
+        if x.ndim == 1: x = x[None, :]
+        improvement = 0
+        dimprovement_dx = 0
+
+        for sample in self.samples:
+            self.model.fix_model_hyperparameters(sample)
+            acquisition = ExpectedImprovement(self.model, self.jitter)
```

Comment:
It feels weird that we need to create `n_samples` new `ExpectedImprovement` objects per evaluation of the acquisition. Is there no way to access the model underneath? I think you could call `self.acquisition.model.fix_model_hyperparameters(sample)` and create the original acquisition in the `__init__`

Reply:
A quick summary of the offline discussion we had with Mark: while possible for EI, it might not be that easy to do for something like EntropySearch, which current implementation carries model-dependent state.

There might be ways around it, but let's push it first and see if performance is a problem at all, with some reasonable benchmark.